api/src/main/java/org/openmrs/module/ModuleFactory.java

```diff
@@ -574,7 +565,7 @@ public static Module getModuleByPackage(String modulePackage) {
 	 * @see Daemon#startModule(Module)
 	 */
 	public static Module startModule(Module module) throws ModuleException {
-		return Daemon.startModule(module);
```

Comment:
I wanted to make sure a modules dependencies are always started before the module is starting. And i thought it would make sense to do that only in one method, that is why i delegated Daemon.startModule(Module) to the one that has my changes. Looking at the implementation of Daemon.startModule(Module) shows that my change makes no difference in the logic because Daemon.startModule(Module) will call Daemon.startModule(Module, false, null) which is essentially the same thing i have done but just at higher level. @dkayiwa does that make sense to you?